File: LeetCode-DSA/Arrays/SortedSquares.py
# Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.
# Q3. Sort squares of numbers in an array in ascending order
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        squared = []
        # Obtain square of number
        for num in nums:
            squared.append((num*num))
        # Sort array using inbuilt function
        squared.sort()
        return squared

File: LeetCode-DSA/Algorithms1/SortedSquares.cpp
// Q977. Squares of sorted array - C++ is faster than Python
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++)
        {
            // Power function returns double value in C++, so we add 0.9 to get number
            nums[i] = (int)(pow(nums[i], 2) + 1e-9);
        }
        // Sort using in built function
        sort(nums.begin(), nums.end());
        return nums;
    }
};